### ISSUE TITLE
Tor connection  crashed after upgrade

### ISSUE THREAD

**Issue by srirambv** [brave/browser-laptop#14399]
### Description
Tor connection  crashed after upgrade

### Steps to Reproduce
  ~~1. Install an older version of beta channel which doesn't include Tor~~
  ~~2. Auto update to current beta version 0.23.8 which includes Tor~~
  ~~3. After upgrade, launch from command line~~
  ~~4. Console throws error about Tor crash~~
  ~~5. Open a Tor tab and load any site, shows connection error page~~
  ~~6. Relaunch from console, might show tor successfully connected (took 2 attempt for me to connect to tor)~~

   1. Launch 0.23.8 from console
   2. Open a tor tab ensure console shows `tor:redy`
   3. Keep the tor tab active/inactive and close browser
   4. Launch browser from console again, shows the tor crashed message
   5. Relaunch browser second time from console, shows `tor:ready` and you can use tor tabs again
 
**Actual result:**
```
$ Crash reporting enabled
tor: control socket error: Error: connect ECONNREFUSED 127.0.0.1:50715
tor: control socket closed early
[18368:16292:0613/164233.610:ERROR:tor_launcher_factory.cc(89)] Tor Process(6532) Crashed
```

**Expected result:**
Should always connect to Tor either on cold launch or on upgrade

**Reproduces how often:**
100%

### Brave Version

**about:brave info:**

Brave | 0.23.8
-- | --
V8 | 6.6.346.32
rev | 06c657b
Muon | 6.1.5
OS Release | 10.0.17134
Update Channel | Beta
OS Architecture | x64
OS Platform | Microsoft Windows
Node.js | 7.9.0
Brave Sync | v1.4.2
libchromiumcontent | 66.0.3359.181

**Reproducible on current live release:**
N/A

**Comment by srirambv** [comment]
This only happens when the browser is closed with an active/inactive private tab. Have to relaunch again for the tor tabs to work.

**Comment by riastradh-brave** [comment]
When you say 'active/inactive', do you mean there's one private tab selected, and another private tab not selected?

Does it make a difference if you wait a few seconds before relaunching?  I wonder whether there's some race condition in killing the process.

Can you share tor.log?  Should be under `%appdata%\brave\tor\`, if I read https://github.com/brave/browser-laptop/wiki/Filesystem-Hierarchy correctly.

**Comment by srirambv** [comment]
@riastradh-brave 
active tab -  is the tab which is in focus
inactive tab - is there on the tab strip but not in focus but a background tab

Here's what the log says 
```
Jun 14 15:38:16.000 [notice] Tor 0.3.2.10 opening new log file.
Jun 14 15:38:16.874 [notice] Tor 0.3.2.10 running on Windows 8 with Libevent 2.0.22-stable, OpenSSL 1.0.2n, Zlib 1.2.8, Liblzma N/A, and Libzstd N/A.
Jun 14 15:38:16.890 [notice] Tor can't help you if you use it wrong! Learn how to be safe at https://www.torproject.org/download/download#warning
Jun 14 15:38:16.894 [notice] Configuration file "C:\nonexistent" not present, using reasonable defaults.
Jun 14 15:38:16.904 [warn] Path for GeoIPFile (<default>) is relative and will resolve to C:\Users\\AppData\Local\BraveBeta\app-0.23.8\<default>. Is this what you wanted?
Jun 14 15:38:16.905 [warn] Path for GeoIPv6File (<default>) is relative and will resolve to C:\Users\\AppData\Local\BraveBeta\app-0.23.8\<default>. Is this what you wanted?
Jun 14 15:38:16.953 [notice] Scheduler type KISTLite has been enabled.
Jun 14 15:38:16.954 [notice] Opening Socks listener on 127.0.0.1:9260
Jun 14 15:38:16.954 [notice] Opening Control listener on 127.0.0.1:0
Jun 14 15:38:16.954 [notice] Control listener listening on port 52640.
Jun 14 15:38:16.000 [warn] Error replacing "C:\Users\\AppData\Roaming\brave-beta\tor\watch\control_auth_cookie": File exists
Jun 14 15:38:16.000 [warn] Error writing auth cookie to "C:\\Users\\AppData\\Roaming\\brave-beta\\tor\\watch\\control_auth_cookie".
Jun 14 15:38:16.000 [warn] Error creating control cookie authentication file.
Jun 14 15:38:16.000 [err] set_options(): Bug: Acting on config options left us in a broken state. Dying. (on Tor 0.3.2.10 )
```

Waiting for about 30 seconds before launch caused the same issue

**Comment by riastradh-brave** [comment]
Candidate fix here: https://github.com/brave/browser-laptop/pull/14426

**Comment by diracdeltas** [comment]
closing due to #14426 being merged into tor/0.23.x; please re-open if it's happening still in the next build

**Comment by srirambv** [comment]
Verified on Windows 10 x64 using
* 0.23.16 - `2042c45`
* Muon - `7.1.1`
* libchromiumcontent - `67.0.3396.87`
